### PR TITLE
Regenerate darwin,linux cgo flags

### PR DESCRIPTION
**Pull request by tamird**

### CODE REVIEW

File: cgo_flags.go [hunk 1]
```diff
@@ -7,14 +7,10 @@ package rocksdb
 // #cgo CPPFLAGS: -Iinternal/utilities/merge_operators/string_append
 // #cgo windows CPPFLAGS: -DOS_WIN
 // #cgo !windows CPPFLAGS: -DROCKSDB_PLATFORM_POSIX -DROCKSDB_LIB_IO_POSIX
-// #cgo darwin CPPFLAGS: -DOS_MACOSX
-// #cgo linux CPPFLAGS: -DOS_LINUX -fno-builtin-memcmp -DROCKSDB_FALLOCATE_PRESENT -DROCKSDB_MALLOC_USABLE_SIZE
+// #cgo darwin CPPFLAGS: -DOS_MACOSX -DROCKSDB_BACKTRACE
+// #cgo linux CPPFLAGS: -DOS_LINUX -fno-builtin-memcmp -DROCKSDB_FALLOCATE_PRESENT -DROCKSDB_MALLOC_USABLE_SIZE -msse -msse4.2
```
Comment (petermattis) [inline cgo_flags.go]: Why did you move the `{-msse,-msse4.2}` flags here? I would think they should be conditional on `amd64` as they were before.

Comment (tamird) [reply cgo_flags.go]: I've replaced these flags with `-march=native` which will enable all supported instruction sets on the host machine (and this is what build_detect_platform emits in the absence of USE_SSE).

Comment (petermattis) [reply cgo_flags.go]: Can you verify the sse code is getting compiled? The easiest way is to edit `internal/util/crc32c.c` and add a #error to the `#ifndef __SSE4_2__` code path.

Comment (tamird) [reply cgo_flags.go]: Yep, this diff produced an error:
```diff
diff --git a/internal/util/crc32c.cc b/internal/util/crc32c.cc
index ce57454..b8e3b1c 100644
--- a/internal/util/crc32c.cc
+++ b/internal/util/crc32c.cc
@@ -14,6 +14,7 @@
 
 #include <stdint.h>
 #ifdef __SSE4_2__
+#error foobar
 #include <nmmintrin.h>
 #endif
 #include "util/coding.h"

```

**Comment by petermattis** [comment]
LGTM
